pkgs/applications/version-management/smartgithg/default.nix

```diff
@@ -1,64 +1,82 @@
-{ stdenv, fetchurl, lib, makeWrapper
-, substituteAll
+{ stdenv
+, fetchurl
+, makeDesktopItem
 , jre
-, gtk2, glib
+, gtk3
+, glib
+, gnome3
+, wrapGAppsHook
+, hicolor-icon-theme
 , libXtst
 , which
 }:
 
 stdenv.mkDerivation rec {
-  name = "smartgithg-${version}";
-  version = "18.2.4";
+  pname = "smartgithg";
+  version = "18.2.9";
 
   src = fetchurl {
     url = "https://www.syntevo.com/downloads/smartgit/smartgit-linux-${builtins.replaceStrings [ "." ] [ "_" ] version}.tar.gz";
-    sha256 = "0ch6vcvndn1fpx05ym9yp2ssfw2af6ac0pw8ssvjkc676zc0jr73";
+    sha256 = "0d08awb2s3w1w8n8048abw2x4wnqhfx96sls9kdsnxj0xrszgz67";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ wrapGAppsHook ];
 
-  buildInputs = [ jre ];
+  buildInputs = [ jre gnome3.adwaita-icon-theme hicolor-icon-theme gtk3 ];
 
-  buildCommand = let
-    pkg_path = "$out/${name}";
-    bin_path = "$out/bin";
-    install_freedesktop_items = substituteAll {
-      inherit (stdenv) shell;
-      isExecutable = true;
-      src = ./install_freedesktop_items.sh;
-    };
-    runtime_paths = lib.makeBinPath [
-      jre
-      #git mercurial subversion # the paths are requested in configuration
-      which
-    ];
-    runtime_lib_paths = lib.makeLibraryPath [
-      gtk2 glib
-      libXtst
-    ];
-  in ''
-    tar xvzf $src
-    mkdir -pv $out
-    mkdir -pv ${pkg_path}
-    # unpacking should have produced a dir named 'smartgit'
-    cp -a smartgit/* ${pkg_path}
-    # prevent using packaged jre
-    rm -r ${pkg_path}/jre
-    mkdir -pv ${bin_path}
-    jre=${jre.home}
-    makeWrapper ${pkg_path}/bin/smartgit.sh ${bin_path}/smartgit \
-      --prefix PATH : ${runtime_paths} \
-      --prefix LD_LIBRARY_PATH : ${runtime_lib_paths} \
+  preConfigure = with stdenv.lib; ''
+    gappsWrapperArgs+=( \
+      --prefix PATH : ${makeBinPath [ jre which ]} \
+      --prefix LD_LIBRARY_PATH : ${makeLibraryPath [
+        gtk3
+        glib
+        libXtst
+      ]} \
       --prefix JRE_HOME : ${jre} \
       --prefix JAVA_HOME : ${jre} \
-      --prefix SMARTGITHG_JAVA_HOME : ${jre}
-    sed -i '/ --login/d' ${pkg_path}/bin/smartgit.sh
-    patchShebangs $out
-    cp ${bin_path}/smartgit ${bin_path}/smartgithg
+      --prefix SMARTGITHG_JAVA_HOME : ${jre} \
+    ) \
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    sed -i '/ --login/d' bin/smartgit.sh
+    mkdir -pv $out/{bin,share/icons,share/applications}
+    cp -av ./{dictionaries,lib} $out/
+    cp -av bin/smartgit.sh $out/bin/smartgit
+    ln -sfv $out/bin/smartgit $out/bin/smartgithg
+    cp -av $desktopItem/share/applications/* $out/share/applications/
+    cp -av bin/*.{png,svg} $out/share/icons/
```

Comment:
These `png` need to be installed to paths like
`share/icons/hicolor/16x16/apps/smartgit.png`

I'd adapt something like
* https://github.com/NixOS/nixpkgs/blob/38b43eeed4326fb1b93d8388d405ac04f7eef809/pkgs/applications/networking/instant-messengers/zoom-us/default.nix#L89

And the svg should be installed to `share/icons/hicolor/scalable/apps/smartgit.svg`.